go-controller/pkg/controller/controller.go

```diff
@@ -0,0 +1,217 @@
+package controller
+
+import (
+	"fmt"
+	"sync"
+
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/factory"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/util"
+
+	"k8s.io/apimachinery/pkg/labels"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/util/workqueue"
+	"k8s.io/klog/v2"
+)
+
+const maxRetries = 15
+
+// Controller is a level-driven controller that is shut down after Stop() call.
+type Controller interface {
+	Start(threadiness int) error
+	Stop()
+	ReconcileAll()
```

Comment:
nit: The `ReconcileAll` looks a bit out of place now. But I guess this should evolve to have a `Reconcile` as  well as some other methods enable watching different types of resources so not bad from that perspective.

Reply:
I expected ReconcileAll() to be moved to the resource handler interface, when one controller will allow multiple resources, so that you can specify which resource should be reconciled, if that makes sense